zebra-test/src/service_extensions.rs

```diff
@@ -0,0 +1,33 @@
+//! Extension traits for [`Service`] types to help with testing.
+
+use futures::future::{BoxFuture, FutureExt};
+use tower::{Service, ServiceExt};
+
+use now_or_later::NowOrLater;
+
+/// An extension trait to check if a [`Service`] is immediately ready to be called.
+pub trait IsReady<Request>: Service<Request> {
+    /// Check if the [`Service`] is immediately ready to be called.
+    fn is_ready(&mut self) -> BoxFuture<bool>;
+
+    /// Check if the [`Service`] is not immediately ready because it returns an error.
+    fn not_ready_due_to_error(&mut self) -> BoxFuture<bool>;
+}
+
+impl<S, Request> IsReady<Request> for S
```

Comment:
Can you also add a method that explicitly checks `is_pending`?

Reply:
Done.